src/tideliftMeUp.ts

```diff
@@ -37,9 +38,22 @@ export async function tideliftMeUp({
 	const allUserPackages = await npmUsernameToPackages(username);
 
 	if (!allUserPackages.length) {
-		throw new TideliftMeUpError(
-			`No packages found for npm username: ${username}.`,
-		);
+		try {
+			await npmUser(username);
+			throw new TideliftMeUpError(
+				`No packages found for npm username: ${username}.`,
+			);
+		} catch (error) {
+			if (
+				typeof error === "object" &&
+				error !== null &&
+				"code" in error &&
+				error.code === "ERR_NO_NPM_USER"
+			) {
+				throw new TideliftMeUpError(`Invalid npm username.`);
+			}
+			throw error;
+		}
```

Comment:
[Refactor] I found this control flow a little confusing. When the `npmUser` call doesn't reject, it throws an error - then immediately catches it, does some checks that won't run true, and re-throws it?

Proposal: how about moving it to after the `try`/`catch`?

I.e.:

```ts
try {
  await npmUser(username);
} catch (error) {
  if ( ... ) {
    throw new TideliftMeUpError(`Invalid npm username.`);
  }
  throw error;
}

throw new TideliftMeUpError(`No packages found for npm username: ${username}.`);
```